### ISSUE TITLE
Add Support for Esri Leaflet Renderers

### ISSUE THREAD

**Issue by pmacMaps** [slutske22/react-esri-leaflet#7]
Are there plans to support Feature Layers utilizing the symbology they are published with?  

This is taken care of the [Esri Leaflet Renderers](https://github.com/Esri/esri-leaflet-renderers) plugin in non-React environments.

**Comment by slutske22** [comment]
I've never played with that plugin before, but now that I look at it, it seems that as long as the renderers plugin is included in your page /code before any esri-leaflet (or react-esri-leaflet) components get added, the renderers plugin just handles the rest for you.  Have you tried this yet?  It seems like it should just work out-of-the-box  I'm a bit busy today, but if you can test it and let me know what the results are, that would help figure out if we actually need to adjust to code to support this.  Otherwise It may take me some time to test it out myself.

**Comment by pmacMaps** [comment]
It appears that simply adding Esri-Leaflet-Renderers throws an error.

![esri-leaflet-renderers-react-error](https://user-images.githubusercontent.com/12861454/122595323-4935b500-d036-11eb-8a61-84597d06d58d.PNG)

**Comment by slutske22** [comment]
How are you adding it?  `<script>` tag in the HTML header?  `import 'esri-leaflet-renderers'`?  Just adding it in shouldn't cause that kind of error.  It may mean you're imports of leaflet, esri-lealfet, and esri-leaflet-renderers are not cooperating with each other.  Can I see some code?

**Comment by slutske22** [comment]
Without seeing some code / attempts to make this work, I'm going to close this issue for the time being.

**Comment by pmacMaps** [comment]
I am adding it as a <script> tag.  It happens whether it's added in the <head> or <body> element.

Here is a zip of sample code:
[test_map.zip](https://github.com/slutske22/react-esri-leaflet/files/6734422/test_map.zip)

**Comment by slutske22** [comment]
Interesting.  Here's the issue:

This is ultimately an inconsistency between including libraries through the html header, vs importing them as a module in a node project.  When including libraries like leaflet from the html header, they get attached to the global window.  So if you follow leaflet's instructions and include it as a `script` tag, you end up with `window.L`, which can then be used in any module.  Similarly, including esri-leaflet in the html header as a script tag attaches `esri` to `window.L`, and you end up with `window.L.esri`.  And one step further, including esri-leaflet-renderers from the html header will try to attach itself to `window.L.esri`.  But in your code example, you only ever included the last of these - esri-leaflet-renderer can't find L or L.esri to latch on to, hence the error.

Most react projects are module-based.  Your project, and this library, are no different. react-esri-leaflet imports leaflet and esri-leaflet directly with statements like `import * as L from 'leaflet'`, or `import * as EL from 'esri-leaflet'`.  That makes them available in the local scope of whatever file they're being imported into.  But they won't necessarily be available at the `window` level.  This is spefically the case for esri-leaflet, when used as a module.  But this is an easy fix - just import esri-leaflet-renderers as a module.

## TLDR - how to fix it

1. Remove`<script src="https://unpkg.com/esri-leaflet-renderers@2.1.2/dist/esri-leaflet-renderers.js" integrity="..."></script>` from your html header
2. `npm install esri-leaflet-renderers`
3. Just after importing anything from esri-leaflet, or react-esri-leaflet, import esri-leaflet-renderers. So in your `Layers.js`:

        import { LayerGroup } from "react-leaflet";
        import { BasemapLayer, FeatureLayer } from "react-esri-leaflet";
        import 'esri-leaflet-renderers'
4. profit:
<img width="1341" alt="Screen Shot 2021-06-29 at 10 01 06 AM" src="https://user-images.githubusercontent.com/52637023/123838520-ee674d80-d8c0-11eb-8c86-bcaa9156b654.png">

Hope that helps, let me know if you have further issues

**Comment by d-belic-levi9-com** [comment]
I'm having the same issue, need it for react-leaflet v2, but tried it in version 3 also, getting the same error in different environments/versions. (esri-leaflet-renderers 2.1.3)

``` TypeError: Cannot read properties of undefined (reading 'addInitHook')```

**Comment by slutske22** [comment]
As I had said in my previous comment, the issue is most likely with a mistmatch of imports.  Without seeing your project code, I can't really help debug.

**Comment by slutske22** [comment]
> I'm having the same issue, need it for react-leaflet v2, but tried it in version 3 also, getting the same error in different environments/versions. (esri-leaflet-renderers 2.1.3)
> 
> ` TypeError: Cannot read properties of undefined (reading 'addInitHook')`

 As I had said in my previous comment, the issue is most likely with a mistmatch of imports.  Without seeing your project code, I can't really help debug.
